lib/generators/react_on_rails/react_with_redux_generator.rb

```diff
@@ -19,33 +24,40 @@ def create_redux_directories
 
       def copy_base_files
         base_js_path = "redux/base"
+        extension = options.typescript? ? "tsx" : "jsx"
 
         # Copy Redux-connected component to auto-registration structure
-        copy_file("#{base_js_path}/app/javascript/bundles/HelloWorld/startup/HelloWorldApp.client.jsx",
-                  "app/javascript/src/HelloWorldApp/ror_components/HelloWorldApp.client.jsx")
-        copy_file("#{base_js_path}/app/javascript/bundles/HelloWorld/startup/HelloWorldApp.server.jsx",
-                  "app/javascript/src/HelloWorldApp/ror_components/HelloWorldApp.server.jsx")
+        copy_file("#{base_js_path}/app/javascript/bundles/HelloWorld/startup/HelloWorldApp.client.#{extension}",
+                  "app/javascript/src/HelloWorldApp/ror_components/HelloWorldApp.client.#{extension}")
+        copy_file("#{base_js_path}/app/javascript/bundles/HelloWorld/startup/HelloWorldApp.server.#{extension}",
+                  "app/javascript/src/HelloWorldApp/ror_components/HelloWorldApp.server.#{extension}")
         copy_file("#{base_js_path}/app/javascript/bundles/HelloWorld/components/HelloWorld.module.css",
                   "app/javascript/src/HelloWorldApp/components/HelloWorld.module.css")
 
         # Update import paths in client component
-        ror_client_file = "app/javascript/src/HelloWorldApp/ror_components/HelloWorldApp.client.jsx"
+        ror_client_file = "app/javascript/src/HelloWorldApp/ror_components/HelloWorldApp.client.#{extension}"
         gsub_file(ror_client_file, "../store/helloWorldStore", "../store/helloWorldStore")
         gsub_file(ror_client_file, "../containers/HelloWorldContainer",
                   "../containers/HelloWorldContainer")
       end
 
       def copy_base_redux_files
         base_hello_world_path = "redux/base/app/javascript/bundles/HelloWorld"
+        component_extension = options.typescript? ? "tsx" : "jsx"
+
+        # Copy non-component files (keep as .js for now)
         %w[actions/helloWorldActionCreators.js
            containers/HelloWorldContainer.js
            constants/helloWorldConstants.js
            reducers/helloWorldReducer.js
-           store/helloWorldStore.js
-           components/HelloWorld.jsx].each do |file|
+           store/helloWorldStore.js].each do |file|
```

Comment:
```suggestion
           store/helloWorldStore.js
           components/HelloWorld.#{component_extension}].each do |file|
```
with `%W` instead of `%w` should work (and remove the added `copy_file` below in that case).